openlibrary/plugins/openlibrary/js/modals/index.js

```diff
@@ -227,6 +246,16 @@ function addClickListeners($modalLinks, maxWidth) {
     })
 }
 
+function addShareModalClickListeners($modalLinks, maxWidth) {
+    $modalLinks.each(function (_i, modalLinkElement) {
+        $(modalLinkElement).on('click', function () {
+            // Get context, which is attached to the modal content
+            const content = getModalContent($(this));
+            displayDynamicModal(content, maxWidth);
+        });
+    });
+}
+
```

Comment:
I think that we can [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) this code by modifying `displayModal`, and using `addClickListeners` to add the listeners to the share modal.